passkey-types/src/webauthn/attestation.rs

```diff
@@ -554,8 +744,36 @@ pub struct CollectedClientData {
 
     /// This OPTIONAL member contains the inverse of the sameOriginWithAncestors argument value that
     /// was passed into the internal method
-    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[serde(default, serialize_with = "truthiness")]
     pub cross_origin: Option<bool>,
+
+    /// CollectedClientData can be extended in the future, this accounts for unknown keys
+    /// Uses an IndexMap to preserve order of keys for JSON byte serialization
+    #[serde(flatten)]
+    pub unknown_keys: IndexMap<String, serde_json::value::Value>,
+}
+
+fn truthiness<S>(cross_origin: &Option<bool>, ser: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    ser.serialize_bool(cross_origin.filter(|b| *b).is_some())
+}
+
+impl CollectedClientData {
+    /// Uses custom JSON byte serialization referenced in the WebAuthn specification, under
+    /// §5.8.1.1 Serialization
+    pub fn to_json_bytes(&self) -> Vec<u8> {
+        collected_client_data_to_json_bytes(self)
+    }
```

Comment:
**Issue:** This should not be part of teh public API, there should be only one way to serialize.

Reply:
Ok will remove